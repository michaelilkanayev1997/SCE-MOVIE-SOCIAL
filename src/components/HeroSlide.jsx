import React, { useState, useEffect } from "react";
import Button from "../components/Button";
import { Link } from "react-router-dom";
import { A11y, Autoplay, Keyboard, Navigation, Pagination } from "swiper";
import { Swiper, SwiperSlide } from "swiper/react";
import apiConfig from "../api/apiConfig";
import "swiper/swiper-bundle.css";
import "../scss/hero-slide.scss";
import { getMediaList } from "../api/axiosClient";
import GlobalLoading from "./GlobalLoading";
import { UserAuth } from "../context/AuthContext";
import UserRating from "./UserRating";
import { OutlineButton } from "../components/Button";

const HeroSlide = () => {
  const [popularMovies, setPopularMovies] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const fetchMovies = async () => {
      setLoading(true);
      const response = await getMediaList("movie", "popular");
      setPopularMovies(response);
      setLoading(false);
    };
    fetchMovies();
  }, []);

  return (
    <>
      {loading ? (
        <GlobalLoading />
      ) : (
        <div className="hero-slide">
          <Swiper
            modules={[Keyboard, Pagination, Navigation, Autoplay, A11y]}
            pagination={{ clickable: true }}
            navigation={true}
            grabCursor={true}
            spaceBetween={0}
            slidesPerView={1}
            keyboard={{
              enabled: true,
            }}
            style={{
              "--swiper-pagination-color": "#FFBA08",
              "--swiper-pagination-bullet-inactive-color": "#999999",
            }}
            //autoplay={{ delay: 3000 }}
          >
            {popularMovies &&
              popularMovies.map((item, i) => (
                <SwiperSlide key={i}>
                  {({ isActive }) => (
                    <HeroSlideItem
                      item={item}
                      className={`${isActive ? "active" : ""}`}
                    />
                  )}
                </SwiperSlide>
              ))}
          </Swiper>
        </div>
      )}
    </>
  );
};

const HeroSlideItem = (props) => {
  const item = props.item;
  const { user } = UserAuth() ?? {};
  const background = apiConfig.originalImage(
    item.backdrop_path ? item.backdrop_path : item.poster_path
  );

  return (
    <div
      className={`hero-slide__item ${props.className}`}
      style={{ backgroundImage: `url(${background})` }}
    >
      <div className="hero-slide__item__content container">
        <div className="hero-slide__item__content__info">
          <h2 className="title">{item.title}</h2>
          <div className="overview">{item.overview}</div>
          <div className="btns">
            {user ? (
              <Button>
                <Link
                  className="property-btn"
                  to="/detail"
                  state={{ item: item, MediaType: "movie" }}
                >
                  Read More
                </Link>
              </Button>
            ) : (
              <OutlineButton>
                <Link className="property-btn" to="/login">
                  sign in to read More
                </Link>
              </OutlineButton>
            )}
          </div>
        </div>

        <div
          className="hero-slide__item__content__poster"
          style={{
            display: "flex",
            flexDirection: "column",
          }}
        >
          <img src={apiConfig.w500Image(item.poster_path)} alt="" />
          <div
            className="user-rating"
            style={{ transform: "scale(1.2)", paddingTop: "18px" }}
          >
            <UserRating
              readOnly={true}
              showButton={false}
              user={user}
              MediaType={"movie"}
              item={item}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default HeroSlide;
