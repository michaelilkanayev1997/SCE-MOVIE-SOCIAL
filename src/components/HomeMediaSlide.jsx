import React, { useState, useEffect } from "react";
import { SwiperSlide, Swiper } from "swiper/react";
import { getMediaList } from "../api/axiosClient";
import MovieCard from "./MovieCard";
import "../scss/movie-slide.scss";

const HomeMediaSlide = (props) => {
  const [items, setItems] = useState([]);

  useEffect(() => {
    const getMovies = async () => {
      window.scrollTo(0, 0);
      const res = await getMediaList(props.type, props.MediaName);

      setItems(res);
    };
    getMovies();
  }, [props.type, props.MediaName]);

  return (
    <div className="movie-list">
      <Swiper grabCursor={true} spaceBetween={10} slidesPerView={6}>
        {items.map((item, i) => (
          <SwiperSlide key={i}>
            <MovieCard item={item} MediaType={props.type} />
          </SwiperSlide>
        ))}
      </Swiper>
    </div>
  );
};

export default HomeMediaSlide;
