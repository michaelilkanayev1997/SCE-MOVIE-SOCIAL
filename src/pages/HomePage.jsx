/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from "react";
import HeroSlide from "../components/HeroSlide";
import { Link } from "react-router-dom";
import { OutlineButton } from "../components/Button";
import HomeMediaSlide from "../components/HomeMediaSlide";
import { UserAuth } from "../context/AuthContext";
import PopUp from "../components/PopUp";

const HomePage = () => {
  const { user, isLoading, userIsBlocked } = UserAuth() ?? {};
  const [PopUpTime, setPopUpTime] = useState(true);
  const [timeoutId, setTimeoutId] = useState(null);

  useEffect(() => {
    const id = setTimeout(() => {
      setPopUpTime(false);
    }, 2000);
    setTimeoutId(id);

    return () => {
      clearTimeout(timeoutId);
    };
  }, []);

  return (
    <>
      {!isLoading && !user && !PopUpTime && !userIsBlocked && <PopUp />}
      <HeroSlide />
      <div className="container">
        <div className="section mb-3">
          <div
            className="section__header mb-2"
            style={{
              textDecorationLine: "underline",
              textDecorationColor: "#FFBA08",
              textDecorationThickness: "4px",
              textUnderlineOffset: "0.5rem",
            }}
          >
            <h2>Upcoming Movies</h2>
            <Link to="/movie">
              <OutlineButton className="small">View more</OutlineButton>
            </Link>
          </div>
          <HomeMediaSlide
            key="upcoming-movies"
            type="movie"
            MediaName="upcoming"
          />
        </div>

        <div className="section mb-3">
          <div
            className="section__header mb-2"
            style={{
              textDecorationLine: "underline",
              textDecorationColor: "#FFBA08",
              textDecorationThickness: "4px",
              textUnderlineOffset: "0.5rem",
            }}
          >
            <h2>Top Rated Movies</h2>
            <Link to="/movie">
              <OutlineButton className="small">View more</OutlineButton>
            </Link>
          </div>
          <HomeMediaSlide
            key="top-rated-movies"
            type="movie"
            MediaName="top_rated"
          />
        </div>

        <div className="section mb-3">
          <div
            className="section__header mb-2"
            style={{
              textDecorationLine: "underline",
              textDecorationColor: "#FFBA08",
              textDecorationThickness: "4px",
              textUnderlineOffset: "0.5rem",
            }}
          >
            <h2>Trending TV</h2>
            <Link to="/tv">
              <OutlineButton className="small">View more</OutlineButton>
            </Link>
          </div>
          <HomeMediaSlide key="trending-tv" type="tv" MediaName="popular" />
        </div>

        <div className="section mb-3">
          <div
            className="section__header mb-2"
            style={{
              textDecorationLine: "underline",
              textDecorationColor: "#FFBA08",
              textDecorationThickness: "4px",
              textUnderlineOffset: "0.5rem",
            }}
          >
            <h2>Top Rated TV</h2>
            <Link to="/tv">
              <OutlineButton className="small">View more</OutlineButton>
            </Link>
          </div>
          <HomeMediaSlide key="top-rated-tv" type="tv" MediaName="top_rated" />
        </div>
      </div>
    </>
  );
};

export default HomePage;
