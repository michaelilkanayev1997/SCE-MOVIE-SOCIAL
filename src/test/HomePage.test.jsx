import HomePage from "../pages/HomePage";
import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import React from "react";
import HeroSlide from "../components/HeroSlide";
import HomeMediaSlide from "../components/HomeMediaSlide";
import "jest-extended";

configure({ adapter: new Adapter() });

describe("HomePage", () => {
  it("should render the page without crashing", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the Upcoming Movies section with a View more button", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find(".section__header h2").at(0).text()).toEqual(
      "Upcoming Movies"
    );
    expect(wrapper.find(".section__header Link").at(0).prop("to")).toEqual(
      "/movie"
    );
    expect(
      wrapper.find(".section__header Link OutlineButton").at(0).text()
    ).toEqual("<OutlineButton />");
  });

  it("renders HeroSlide", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find(HeroSlide)).toHaveLength(1);
  });

  it("renders Upcoming Movies section", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find("h2").at(0).text()).toEqual("Upcoming Movies");
    expect(wrapper.find(HomeMediaSlide).at(0).prop("type")).toEqual("movie");
    expect(wrapper.find(HomeMediaSlide).at(0).prop("MediaName")).toEqual(
      "upcoming"
    );
  });

  it("renders Top Rated Movies section", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find("h2").at(1).text()).toEqual("Top Rated Movies");
    expect(wrapper.find(HomeMediaSlide).at(1).prop("type")).toEqual("movie");
    expect(wrapper.find(HomeMediaSlide).at(1).prop("MediaName")).toEqual(
      "top_rated"
    );
  });

  it("renders Trending TV section", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find("h2").at(2).text()).toEqual("Trending TV");
    expect(wrapper.find(HomeMediaSlide).at(2).prop("type")).toEqual("tv");
    expect(wrapper.find(HomeMediaSlide).at(2).prop("MediaName")).toEqual(
      "popular"
    );
  });

  it("renders Top Rated TV section", () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find("h2").at(3).text()).toEqual("Top Rated TV");
    expect(wrapper.find(HomeMediaSlide).at(3).prop("type")).toEqual("tv");
    expect(wrapper.find(HomeMediaSlide).at(3).prop("MediaName")).toEqual(
      "top_rated"
    );
  });
});
