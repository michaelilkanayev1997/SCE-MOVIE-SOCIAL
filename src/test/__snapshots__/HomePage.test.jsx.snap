// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render the page without crashing 1`] = `ShallowWrapper {}`;
